remotespark/kernels/kernelmagics.py

```diff
@@ -264,9 +285,12 @@ def _do_not_call_start_session(self, line, cell="", local_ns=None):
         return self.session_started
 
     @cell_magic
+    @handle_expected_exceptions
     def _do_not_call_delete_session(self, line, cell="", local_ns=None):
-        if self.session_started:
-            self.spark_controller.delete_session_by_name(self.session_name)
+        try:
+            if self.session_started:
+                self.spark_controller.delete_session_by_name(self.session_name)
+        finally:
```

Comment:
>>> try:
    ...   raise ValueError('hi')
    ... finally:
    ...   print 'yo'
    ...
    yo
    Traceback (most recent call last):
      File "<stdin>", line 2, in <module> 
    ValueError: hi